elyra/metadata/schemas/kfp.json

```diff
@@ -53,6 +54,34 @@
             "category": "Kubeflow Pipelines"
           }
         },
+        "engine": {
+          "title": "Kubeflow Pipelines engine",
+          "description": "The Kubeflow Pipelines engine in use",
+          "type": "string",
+          "enum": ["Argo", "Tekton"],
+          "default": "Argo",
+          "uihints": {
+            "field_type": "dropdown",
+            "category": "Kubeflow Pipelines"
+          }
+        },
+        "auth_type": {
+          "title": "Authentication Type",
+          "description": "Authentication type Elyra uses to authenticate with Kubeflow",
+          "type": "string",
+          "enum": [
+            "DEX (generic)",
+            "DEX with LDAP",
+            "DEX with static passwords",
+            "Kubernetes service account token",
+            "No authentication"
+          ],
```

Comment:
Hmm - these will contradict my other review comment about preserving the level of indirection to the displayable value.  If we use the enum _name_ fields here, we are free to change the displayable values.  Of course, I'm not sure where _values_ will be displayed other than in the metadata editor.  Not sure how best to handle enums in a metadata-driven app.

One approach might be to _build_ a `uihint`via the Runtimes schemaspace implementation that looks for auth_type and essentially puts a mapping from the enum's `name` to its `value`.  The metadata editor persists the field using the `name` but displays the `value` when such a hint is present (if that makes sense).

cc: @marthacryan

Reply:
@ptitzler similar to my [comments here](https://github.com/elyra-ai/elyra/pull/2257/files#r740680262), I think the enum should be strings without whitespace in them:
- `NO_AUTHENTICATION`
- `K8S_SERVICEACCOUNT_TOKEN`
- `DEX_LDAP_CONNECTOR`
- `DEX_STATIC_PASSWORDS`
- `DEX_LEGACY`

Reply:
Hi Mathew - this is an area where being metadata-driven can be a bit thorny.  On the one hand, we'd like to present users with nice displayable (potentially localizable) values, while, on the other, introduce a level of indirection for the _persisted_ values that essentially remain fixed and are not privy to the whims of others inherent in human display values.  This is exactly what is driving my previous _`name_value_map`_ comment.

To elaborate more on the proposed `uihints`, I think this schema stanza would resemble something like the following:
```JSON
        "auth_type": {
          "title": "Authentication Type",
          "description": "Authentication type Elyra uses to authenticate with Kubeflow",
          "type": "string",
          "enum": [
            "DEX_LEGACY",
            "DEX_LDAP_CONNECTOR",
            "DEX_STATIC_PASSWORDS",
            "K8S_SERVICEACCOUNT_TOKEN",
            "NO_AUTHENTICATION"
          ],
          "default": "No authentication",
          "uihints": {
            "field_type": "dropdown",
            "category": "Kubeflow Pipelines",
            "name_value_map": {
                "NO_AUTHENTICATION": "No authentication",
                "DEX_STATIC_PASSWORDS": "DEX with static passwords",
                "K8S_SERVICEACCOUNT_TOKEN": "Kubernetes service account token",
                "DEX_LEGACY": "DEX (generic)",
                "DEX_LDAP_CONNECTOR": "DEX with LDAP"
            }
          }
        },
```
The metadata-editor would be responsible for converting schema `enum` values to their displayable value found in the map **and** setting the field to the selected _name_ (constant-looking string) value in order to adhere to the constraints imposed by the _enum_ (so schema validation succeeds).

If we find we can't implement a `uihints.name_value_map` in this release, then we let the constant-looking values ride.

Reply:
The latest updates:
 - use the `SupportedAuthProviders` enum to automatically populate the list of supported authentication types in the GUI
 - display the enum member name instead of the enum member value
 - use enum members wherever possible
 - should address all enum related feedback